### PR TITLE
Format assert statements.

### PR DESCRIPTION
**Pull request by kallentu**
This looks really similar to how asserts were formatted in the old formatter.
It's also similar logic to assert initializers, but we need constructors done first and then I'll handle that after.

I assume we have test logic for comments in ListPieces and general statements so I didn't add any extra for that, but if you think otherwise, let me know!

### CODE REVIEW

**Comment by kallentu** [comment]
Oh wait, before reviewing, let me push another commit. I forgot to push this one.

**Comment by kallentu** [comment]
Okay, should be ready for review now!

File: lib/src/front_end/ast_node_visitor.dart [hunk 1]
```diff
@@ -130,7 +130,17 @@ class AstNodeVisitor extends ThrowingAstVisitor<void>
...
-    throw UnimplementedError();
+    token(node.assertKeyword);
+
+    var arguments = [node.condition];
+    if (node.message case var message?) arguments.add(message);
+    createList(
+      arguments,
+      leftBracket: node.leftParenthesis,
+      rightBracket: node.rightParenthesis,
+    );
```
Comment (munificent) [inline lib/src/front_end/ast_node_visitor.dart:141]: If you really wanted to be clever, you could do:

```dart
    createList(
      [
        node.condition,
        if (node.message case var message?) message
      ],
      leftBracket: node.leftParenthesis,
      rightBracket: node.rightParenthesis,
    );
```

**Review by munificent — APPROVED** [review]
This is great!

File: lib/src/front_end/ast_node_visitor.dart (hunk 1 above)
Comment (kallentu) [reply lib/src/front_end/ast_node_visitor.dart:141]: Oh that's kinda wild. Done.

**Pull request merged**